src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/Level.java

```diff
@@ -199,19 +199,19 @@ public void closeNextNode(final int rotation)
                 return;
         }
 
-        for (final Point2D.Double pos : nodeCenterList)
+        for (final Vec2i pos : nodeCenterList)
         {
             if (nodes.containsKey(pos))
             {
                 continue;
             }
-            final Node tempNodeToAdd = new Node(pos.getX(), pos.getY(), new Point2D.Double(tempNode.getX(), tempNode.getZ()));
+            final Node tempNodeToAdd = new Node(pos.getX(), pos.getZ(), new Vec2i(tempNode.getX(), tempNode.getZ()));
             final int randNumber = rand.nextInt(RANDOM_TYPES);
             tempNodeToAdd.setStyle(randNumber <= 1 ? TUNNEL : (randNumber == 2 ? BEND : CROSSROAD));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)